### PR TITLE
MQE-588: CreatedData should throw Error/Warning when undefined data i…

### PR DESCRIPTION
**Pull request by jilu1**
…s returned

### Fixed Issues (if relevant)

1. magento/magento2-functional-testing-framework#<issue_number>: Issue title
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

**Comment by soumyau** [comment]
I see build failures in - https://m2build-ur.devops.magento.com/job/All-User-Requested-Tests/23345/ due to expections like this: `Undefined field default_value in entity object with a stepKey of createConfigProductAttribute`. Please verify

File: dev/tests/unit/Magento/FunctionalTestFramework/DataGenerator/Handlers/PersistedObjectHandlerTest.php [hunk 1]
```diff
@@ -416,10 +428,15 @@ public function testRetrieveEntityInValidField($name, $key, $value, $type, $scop
...
         $handler->createEntity($stepKey, $scope, $name);
 
-        $this->expectException(\Magento\FunctionalTestingFramework\Exceptions\TestReferenceException::class);
-
         // Call method
         $handler->retrieveEntityField($stepKey, $invalidDataKey, $scope);
+
+        // validate log statement
+        TestLoggingUtil::getInstance()->validateMockLogStatement(
+            "warning",
```
Comment (soumyau) [inline dev/tests/unit/Magento/FunctionalTestFramework/DataGenerator/Handlers/PersistedObjectHandlerTest.php]: maybe change double quotes to single? Not the end of the world, but read that it avoids processing overhead for strings.

**Comment by jilu1** [comment]
> I see build failures in - https://m2build-ur.devops.magento.com/job/All-User-Requested-Tests/23345/ due to expections like this: `Undefined field default_value in entity object with a stepKey of createConfigProductAttribute`. Please verify

Updated mainline and get green build https://m2build-ur.devops.magento.com/job/All-User-Requested-Tests/23527/

File: dev/tests/unit/Magento/FunctionalTestFramework/DataGenerator/Handlers/PersistedObjectHandlerTest.php (hunk 1 above)
Comment (jilu1) [reply dev/tests/unit/Magento/FunctionalTestFramework/DataGenerator/Handlers/PersistedObjectHandlerTest.php]: Fixed

**Pull request merged**